File: src/test/util/EchoSocket1.rb
require 'socket'

clientIn = TCPSocket.new 'localhost', ENV["STDIN_PORT"].to_i #8991
clientOut = TCPSocket.new 'localhost', ENV["STDOUT_PORT"].to_i #8990

puts "Connection is opened"

begin
  while  msg = clientIn.recv( 1024 )  do # Read message from socket
    puts msg
    clientOut.send msg, 0
  end
rescue SystemExit, Interrupt
  raise
rescue Exception => e
  puts "EOF........"
  clientIn.close_read             # close socket when done
  clientOut.close_write
end  

File: src/test/util/EchoSocket1.js
let net = require('net');
console.log("EchoSocket1 start");
try {
	let clientIn = new net.Socket();
	let clientOut = new net.Socket();

	clientOut.connect(process.env.STDOUT_PORT, '127.0.0.1', function () {
		console.log('Connected out');
	});
	clientIn.connect(process.env.STDIN_PORT, '127.0.0.1', function () {
		console.log('Connected in');
	});
	clientIn.on('data', function (data) {
		console.log('Received: ' + data);
		clientOut.write(data);
	});

	//TODO probably remove
	setInterval(function () {
		clientOut.write("");
	}, 1000);

} catch (e) {
	console.log("EchoSocket1 issue: " + e);
}